test/src/main/java/org/apache/accumulo/test/rpc/ThriftBehaviorIT.java

```diff
@@ -0,0 +1,191 @@
+import org.junit.rules.Timeout;
+
+@Category(SunnyDayTests.class)
+public class ThriftBehaviorIT {
+
+  @Rule
+  public Timeout timeout = new Timeout(5, TimeUnit.SECONDS);
+
+  @Rule
+  public TestName testName = new TestName();
+
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
+
+  private SimpleThriftService.Client client;
+  private SimpleThriftServiceHandler handler;
+  private SimpleThriftServiceRunner serviceRunner;
+  private String propName;
+  private Map<Logger,Level> oldLogLevels = new HashMap<>();
+
+  private static final String KITTY_MSG = "🐈 Kitty! 🐈";
+
+  // can delete wrapper when tests pass without using it (assuming tests are good enough)
+  private static final boolean USE_RPC_WRAPPER = true;
+
+  private static final boolean SUPPRESS_SPAMMY_LOGGERS = true;
+
+  @Before
+  public void createClientAndServer() {
+    Arrays.stream(new Class<?>[] {TSimpleServer.class, ProcessFunction.class}).forEach(spammyClass -> {
+      Logger spammyLogger = Logger.getLogger(spammyClass);
+      oldLogLevels.put(spammyLogger, spammyLogger.getLevel());
+      if (SUPPRESS_SPAMMY_LOGGERS) {
+        spammyLogger.setLevel(Level.OFF);
+      }
+    });
+
+    String threadName = ThriftBehaviorIT.class.getSimpleName() + "." + testName.getMethodName();
+    serviceRunner = new SimpleThriftServiceRunner(threadName, USE_RPC_WRAPPER);
+    serviceRunner.startService();
+    client = serviceRunner.client();
+    handler = serviceRunner.handler();
+
+    propName = testName.getMethodName();
+    if (propName.endsWith("Handler")) {
+      propName = propName.substring(0, propName.length() - 7);
+    }
+    propName = SimpleThriftServiceHandler.class.getSimpleName() + "." + propName;
+
+    System.setProperty(propName, "-");
+    Assert.assertEquals("-", System.getProperty(propName));
+  }
+
+  @After
+  public void shutdownServer() {
+    serviceRunner.stopService();
+
+    oldLogLevels.forEach((spammyLogger, oldLevel) -> {
+      spammyLogger.setLevel(oldLevel);
+    });
+
+    Assert.assertEquals(KITTY_MSG, System.getProperty(propName));
+  }
+
+  @Test
+  public void echoFailHandler() throws TException {
+    exception.expect(TException.class);
+    exception.expectCause(IsInstanceOf.instanceOf(UnsupportedOperationException.class));
+    handler.echoFail(KITTY_MSG);
+  }
+
+  @Test
+  public void echoFail() throws TException {
+    try {
+      client.echoFail(KITTY_MSG);
+      Assert.fail();
+    } catch (Exception e) {
+      Assert.assertEquals(TApplicationException.class.getName(), e.getClass().getName());
+    }
+    // verify normal two-way method still passes using same client
+    echoPass();
+  }
+
+  @Test
+  public void echoRuntimeFailHandler() throws TException {
+    exception.expect(UnsupportedOperationException.class);
+    handler.echoRuntimeFail(KITTY_MSG);
+  }
+
+  @Test
+  public void echoRuntimeFail() throws TException {
+    try {
+      client.echoRuntimeFail(KITTY_MSG);
+      Assert.fail();
```

Comment:
I suppose I can add a message, but the name of the method and the being wrapped by a try-catch make it pretty obvious.